modules/consensus/consensusset.go

```diff
@@ -222,6 +228,12 @@ func (cs *ConsensusSet) CurrentBlock() (block types.Block) {
 	return block
 }
 
+// Flush will block until the consensus set has finished all in-progress
+// routines.
+func (cs *ConsensusSet) Flush() error {
+	return cs.tg.Flush()
+}
```

Comment:
You can spin on `cs.CurrentBlock` to wait for the block. Or, if you want cleaner synchronization, subscribe to the consensus set:
```go
type subscriberFn func(modules.ConsensusChange)

func (fn subscriberFn) ProcessConsensusChange(cc modules.ConsensusChange) {
    fn(cc)
}

func TestFoo() {
    cs := consensusSet.New()
    blockNotify := make(chan struct{})
    cs.Subscribe(subscriberFn(func(modules.ConsensusChange) {
        blockNotify <- struct{}{}
    }))
    // wait for block
    <-blockNotify
}
```
The above is probably insufficient because you need a corresponding channel read for each block, but with a few tweaks I'm sure it could serve your purposes.